enhancements/network/persistent-ips-sdn-secondary-networks.md

```diff
@@ -0,0 +1,1012 @@
+garbage collector is the entity responsible for deleting the persistent IP
+allocation.
+
+```mermaid
+sequenceDiagram
+    actor user
+
+    user ->> apiserver: Delete(vm)
+    apiserver ->> gc: Delete(vm)
+    gc->>apiserver: Delete(virtLauncherPod)
+    apiserver-->>gc: Result
+    gc->>apiserver: Delete(ipamLease)
+    apiserver-->>gc: Result
+    apiserver->>ipamLeaseOperator: Reconcile(ipamLease)
+    ipamLeaseOperator ->> ipamLeaseOperator: DeallocateIPs(ipamLease.Status.IPs)
+```
+
+#### OVN-Kubernetes
+OVN-Kubernetes changes would be pretty minimal: it already supports setting
+static IP addresses via `NetworkSelectionElements`, but **only** when the
+subnet parameter is omitted. We would need to change that, allowing
+OVN-Kubernetes to honor IP requests from their pools.
+
+#### New IPAM controller
+This IPAM controller would need to behave as the *real* IPAM component; once it
+sees an `IPAMLease` request, it would need to generate a new IP for the
+requesting pod, and update the `IPAMLease` with it.
+
+High availability would be required, meaning this extra component would need to
+be a deployment of at least 2 pods (active / backup), to ensure VMs with
+persistent IPs could be serviced even if one of the IPAM lease controllers
+crashes.
+
+#### Advantages of this alternative
+
+The OVN-Kubernetes existing IPAM solution would only require a minor tweak: we
+would need to allow static IP requests on NADs with a defined subnet attribute.
+
+This solution would be generic: other workloads interested in this feature
+would just have to create the CR, and wait until an IP is reported in its
+status, then template the pod accordingly.
+
+Also, the API on how to use the persistent IPs feature would be very clean: the
+client interested in having persistent IPs for their workloads simply
+provisions the `IPAMLease` objects, the IPAM CNI updates their status
+sub-resource with the chosen addresses, and the client controller would then
+template the pod featuring the IP request in the pod's network selection
+elements.
+
+#### Disadvantages of this alternative
+
+- KubeVirt would have to "track" a new CRD.
+- KubeVirt memory footprint would increase
+- Reinventing the IPAM wheel
+- Productifying a new IPAM solution takes time
+- Shipping / release engineering a new component
+- We're basically implementing IPAM outside CNI. Clunky flow.
+
+#### Mitigation of the alternative's drawbacks
+
+Instead of having a separate controller, the functionality described above
+could be shipped in OVN-Kubernetes, as a separate flow. This would remove most
+of the disadvantages of the alternative.
+
+This approach would still have one inconvenient - it would take longer to
+create a VM, since the virt-controller would first need to wait for the IPAM
+plugin to notify the persistent IP allocation, and only then would be able to
+template the pod.
+
+#### Mitigation the added IP reporting time
```

Comment:
The last time we talked about it I understood this is the preferred solution. However in the doc it is just an alternative. I"m a bit confused.

Reply:
Done.